pkg/cloud/mesos/config.go

```diff
@@ -0,0 +1,79 @@
+package mesos
+
+import (
+	"io"
+	"time"
+
+	"code.google.com/p/gcfg"
+)
+
+const (
+	DefaultMesosMaster       = "localhost:5050"
+	DefaultHttpClientTimeout = time.Duration(10) * time.Second
+	DefaultStateCacheTTL     = time.Duration(5) * time.Second
+)
+
+var (
+	config *Config
+)
+
+func getConfig() *Config {
+	return config
+}
+
+func resetConfigToDefault() {
+	config = createDefaultConfig()
+}
+
+func init() {
+	resetConfigToDefault()
+}
+
+// Example Mesos cloud provider configuration file:
+//
+// [mesos-cloud]
+//  mesos-master        = leader.mesos:5050
+//	http-client-timeout = 500ms
+//	state-cache-ttl     = 1h
+
+type ConfigWrapper struct {
+	Mesos_Cloud Config
+}
+
+type Config struct {
+	MesosMaster            string          `gcfg:"mesos-master"`
+	MesosHttpClientTimeout WrappedDuration `gcfg:"http-client-timeout"`
+	StateCacheTTL          WrappedDuration `gcfg:"state-cache-ttl"`
+}
+
+type WrappedDuration struct {
```

Comment:
Sorry I should have explained myself. We don't want to duplicate all the methods like they do. I was trying to point out that:
```go
type Duration struct {
   time.Duration
}
```
is more idiomatic than:
```go
type WrappedDuration struct {
    Duration time.Duration
}
```

If we want to eliminate the `.Duration` but keep access to funcs on `time.Duration` then we can declare this way:
```go
type Duration struct {
   *time.Duration
}
```

So, now we can say something like this:
```go
func doSomething(d Duration) {
   fmt.Printf("number of seconds in d is %d", d.Seconds())
}
```